Applications/Spire/Source/Ui/DropDownBox.cpp

```diff
@@ -229,21 +238,23 @@ void DropDownBox::on_list_view_current(const boost::optional<int>& current) {
   m_text_box->get_model()->set_current(std::move(text_current));
 }
 
-void DropDownBox::on_list_view_submit(const std::any& submission) {
+void DropDownBox::on_submit(const std::any& submission) {
   m_drop_down_list->hide();
-  update_submission();
+  m_submission = submission;
+  m_submit_signal(m_submission);
 }
 
 void DropDownBox::update_current() {
-  if(m_submission != m_list_view->get_current_model()->get_current()) {
-    m_list_view->get_current_model()->set_current(m_submission);
-  }
+  m_list_view->get_current_model()->set_current(
+    get_index(m_list_view->get_list_model(), displayTextAny(m_submission)));
 }
 
 void DropDownBox::update_submission() {
-  m_submission = m_list_view->get_current_model()->get_current();
-  if(m_submission) {
-    m_submit_signal(m_list_view->get_list_model()->at(*m_submission));
+  auto index = get_index(m_list_view->get_list_model(),
```

Comment:
See above, do not needlessly make the code significantly slower without a good justification for it.